src/databricks/labs/dqx/engine.py

```diff
@@ -1020,3 +1020,133 @@ def _save_checks_in_table(checks: list[dict], table_name: str, run_config_name:
         rules_df.write.option("replaceWhere", f"run_config_name = '{run_config_name}'").saveAsTable(
             table_name, mode=mode
         )
+
+    def apply_checks_and_write_to_table(
+        self,
+        input_table: str,
+        checks: list[DQRule],
+        output_table: str,
+        output_table_mode: str = "append",
+        output_table_options: dict[str, str] | None = None,
+        quarantine_table: str | None = None,
+        quarantine_table_mode: str | None = None,
+        quarantine_table_options: dict[str, str] | None = None,
+        trigger: dict[str, Any] | None = None,
+        spark: SparkSession | None = None,
+    ) -> DataFrame | tuple[DataFrame, DataFrame]:
+        """
+        Apply data quality checks to a table or view and write the result to Delta table(s).
+        
+        If quarantine_table is provided, the data will be split into good and bad records,
+        with good records written to output_table and bad records to quarantine_table.
+        If quarantine_table is not provided, all records (with error/warning columns) 
+        will be written to output_table.
+
+        :param input_table: name of the table or view to read data from
+        :param checks: list of checks to apply to the dataframe. Each check is an instance of DQRule class.
+        :param output_table: name of the output table to save the data
+        :param quarantine_table: optional name of the quarantine table to save bad data. 
+            If provided, data will be split into good/bad records.
+        :param output_table_mode: Output mode for writing to the output table (default is 'append'),
+            not applicable for streaming DataFrames
+        :param quarantine_table_mode: Output mode for writing to the quarantine table (default is 'append'),
+            not applicable for streaming DataFrames
+        :param output_table_options: Additional options for writing to the output table
+        :param quarantine_table_options: Additional options for writing to the quarantine table
+        :param trigger: Trigger options for streaming DataFrames, e.g. {"availableNow": True}
+        :param spark: Optional SparkSession. If not provided, will use SparkSession.builder.getOrCreate()
+        :return: If quarantine_table is provided, returns tuple of (good_df, bad_df).
+            Otherwise, returns dataframe with errors and warning reporting columns.
+        """
+        if spark is None:
+            spark = SparkSession.builder.getOrCreate()
+        
+        # Read data from the specified table
+        df = spark.read.table(input_table)
```

Comment:
Please use `read_input_data` as it handles both paths and tables. The `read_input_data` should be also be extended to support streaming as well.